post-processing/streamlit_post_processing.py

```diff
@@ -102,12 +111,77 @@ def update_config():
 
     state = st.session_state
     uploaded_config = state.uploaded_config
+    df = state.post.df
+
     if uploaded_config:
         try:
             config_dict = load_config(uploaded_config)
             state.config = ConfigHandler(config_dict)
+            config = state.config
```

Comment:
Is all this needed to solve https://github.com/ukri-excalibur/excalibur-tests/issues/345#issuecomment-2517661658 ? How come with streamlit version 1.30.0 it works?

Reply:
Yes, these `update_config` changes solve the exact problem you're describing in that comment. My guess is that in the newer Streamlit versions they must have either changed something about how session state updates or how text input / selectboxes update their values.

It used to be the case that changing the config in the session state was enough to update everything that pulled from it, but now we get a completely different behaviour where the text input and selectbox fields don't update according to the new config and instead change some of the config values back to whatever was originally in the UI fields. I suspect it has something to do with allowing empty / null / None in those fields, as type selectboxes don't seem to have this issue. Either way, it should all be sorted now.

Reply:
Ok, I see. Is it worth it to open an issue in the upstream streamlit repo? I mean, they degraded in functionality...

Reply:
I agree, it was quite tedious to have to write in these additions to avoid breaking the UI when it just worked before, but I don't know why they made the changes they did, or even necessarily what the changes were, so not sure I'd like to open an issue without at least investigating the change logs.

Reply:
Sure, check the logs first. But opening an issue doesn't mean you have to solve it as well, you're doing them a favour by spotting something they didn't.